scripts/set-react-version.mjs

```diff
@@ -417,18 +448,24 @@ export function setReactVersion(version, coreOnly) {
 const { [1]: script, [2]: version } = process.argv;
 if (import.meta.url.endsWith(script)) {
   if (!(await checkEnvironment())) {
-    // eslint-disable-next-line local/no-process-exit
-    process.exit(1);
-  }
-
-  if (!isValidVersion(version)) {
+    process.exitCode = 1;
+  } else if (!isValidVersion(version)) {
     const script = path.basename(fileURLToPath(import.meta.url));
     console.log(
       `Usage: ${script} [<version number> | ${VALID_TAGS.join(" | ")}]`
     );
-    // eslint-disable-next-line local/no-process-exit
-    process.exit(1);
+    process.exitCode = 1;
+  } else {
+    setReactVersion(version, process.argv.includes("--core-only")).then(() => {
+      const numVersion =
+        version === "nightly"
+          ? Number.MAX_SAFE_INTEGER
+          : toVersionNumber(version);
+      if (numVersion >= v(0, 74, 0)) {
+        disableJetifier();
+      } else if (numVersion < v(0, 73, 0)) {
+        setGradleVersion("7.6.3");
```

Comment:
based on my testing for the other PR -> https://github.com/microsoft/react-native-test-app/pull/1837#issuecomment-1941437582

we still have the webstorage webapi problem when downgrading and testing Android; do we want/need to have a patch similar to the one we had for safearea to apply here? we can leave it for a follow up PR if yes